src/Symfony/Component/Console/Helper/Table.php

```diff
@@ -347,13 +371,46 @@ public function render()
                     }
                     if (isset($row[$i])) {
                         $rows[$i][] = $row[$i];
-                    } elseif ($rows[$i][0] instanceof TableCell && $rows[$i][0]->getColspan() >= 2) {
+                    } elseif ($isCellWithColspan($rows[$i][0])) {
                         // Noop, there is a "title"
                     } else {
                         $rows[$i][] = null;
                     }
                 }
             }
+        } elseif ($vertical) {
+            $maxHeaderLength = max(array_map(static function (string $header) {
+                return mb_strlen($header);
+            }, $this->headers[0] ?? ['']));
+
+            foreach ($this->rows as $row) {
+                if ($row instanceof TableSeparator) {
+                    continue;
+                }
+
+                if (0 < \count($rows)) {
+                    $rows[] = [$divider];
+                }
+
+                $containsColspan = 0 < \count(array_filter($row, static function ($cell) use ($isCellWithColspan): bool {
+                    return $isCellWithColspan($cell);
+                }));
+
+                $headers = $this->headers[0] ?? [];
+                $maxRows = max(\count($headers), \count($row));
+                for ($i = 0; $i < $maxRows; ++$i) {
+                    $cell = (string) ($row[$i] ?? '');
+                    if (!empty($headers) && !$containsColspan) {
```

Comment:
`empty` should not be used when possible.
```suggestion
                    if ([] === $headers && !$containsColspan) {
```

Reply:
done